src/Login.tsx

```diff
@@ -59,7 +59,7 @@ class Login extends React.Component<Props, State> {
         super(props)
 
         this.state = {
-            endpoint: "",
+            endpoint: this.props.session.endpoint,
```

Comment:
you should use this getDerivedStateFromProps instead

Reply:
Hi Sylvain, I do not know even the ABC of react :-) just googled about the getDerivedStateFromProps and got this blog https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html , I feel in this case we no need to use derived state. I may wrong also if so correct me, I am happy to add derived state. Thanks!

Reply:
me neither, let's merge like this